### DISCUSSION TITLE
How can I affect the HTML in a layout component based on the result of a request in a page component?

### DISCUSSION THREAD

**Discussion by gavin667** [nuxt/nuxt#24417] [Questions]
I have a page that queries a CMS for a bunch of fields. One of those fields is a theme field. Each page could have a different theme set. My navigation is in a layout file. How do I tell the navigation in the layout file about the theme it should set on initial render without causing a hydration error?

One option is this:

```
     useHead({
       bodyAttrs: {
         class: `is-${pageTheme}-theme`
       }
     })
```

...and use CSS to handle things.

However I want to pass a theme prop to the navigation.

From my understanding you can't affect the layout from a page within it on initial render. Which leads me to the question of how come `useHead` is able to correctly place a class on the body tag ok during server render (which is outside the page component) but I can't affect my own layout during server render?

In Nuxt 2 we could use asyncData and set a value on the store and this would affect the layout correctly on server render but with the setup functions that doesn't seem possible now. We end with hydration mismatches.

What is the equivalent solution in Nuxt 3?

**Comment by darioferderber** [comment]
Hi @gavin667!
It would be much easier if you can provide a [reproduction](https://stackblitz.com/fork/github/nuxt/starter/tree/v3-stackblitz).
My guess is that `pageTheme` is initialized after the `Navigation` component gets mounted.

Example
```vue
<template>
  <Navigation :page-theme="pageTheme" />
  <slot />
</template>
```

If `pageTheme` gets initialized somewhere in the `slot` - this could potentially lead to hydration mismatches.
Best way would be to initialize `pageTheme` in the `app.vue`

Hope this helps 🤞
